### PR TITLE
Fix sourceUrl when there's no replaceParams

### LINKED ISSUE #17671: SOURCE_URL appending extraneous equals sign
## What's the issue?

While using SOURCE_URL within AMP string substitutions, it adds an extra equals sign to params that have no value. For example, if the url is "https://example.com/?amp", the SOURCE_URL variable would evaluate to "https://example.com/?amp="

There is a discrepancy with the value in window.context.sourceUrl (described in https://github.com/ampproject/amphtml/blob/master/ads/README.md) as well. The sourceUrl variable does not have the extra equals sign.

## How do we reproduce the issue?

Can reproduce by visiting https://amp-playground.firebaseapp.com/?amp and using the SOURCE_URL variable

i.e. `  <amp-pixel src="https://foo.com/pixel?href=SOURCE_URL"></amp-pixel>`

screen shot 2018-08-22 at 2 25 41 pm

## What browsers are affected?

Demonstrated through amp playground, chrome dev tools, but observed over many browsers

## Which AMP version is affected?

Powered by AMP ⚡ HTML – Version 1534872626733 https://amp-playground.firebaseapp.com/?amp

### PR DESCRIPTION
**Pull request by zhouyx**
To fix #17671 

cc @brandondiamond

### CODE REVIEW

File: test/functional/test-url-replacements.js [hunk 1]
```diff
@@ -432,15 +432,15 @@ describes.sandboxed('UrlReplacements', {}, () => {
...
       sandbox.stub(trackPromise, 'getTrackImpressionPromise').callsFake(() => {
         return Promise.resolve();
       });
       return Services.urlReplacementsForDoc(win.ampdoc)
           .expandUrlAsync('?url=SOURCE_URL')
           .then(res => {
-            expect(res).to.contain(
-                encodeURIComponent('http://o.com/foo/?hello=world'));
+            expect(res).to.equal(
+                '?url=' + encodeURIComponent('http://o.com/foo/?hello=world&a='));
```
Comment (choumx) [inline test/functional/test-url-replacements.js]: @brandondiamond Is the trailing `=` right?

Comment (zhouyx) [reply test/functional/test-url-replacements.js]: Honestly I think we should stick the the format as best as we can.  But @lannka believes it makes sense to normalize the params if we need to change it anyway by add those extra params. Open to change the logic if there's any ideas.

Comment (brandondiamond) [reply test/functional/test-url-replacements.js]: I don't believe there was a recommendation one way or another when we were chatting about implementing this. As long as we're consistent, we should be fine :+1:

File: test/functional/test-url-replacements.js [hunk 2]
```diff
@@ -454,8 +454,38 @@ describes.sandboxed('UrlReplacements', {}, () => {
...
+      });
+      return Services.urlReplacementsForDoc(win.ampdoc)
+          .expandUrlAsync('?url=SOURCE_URL')
+          .then(res => {
+            expect(res).to.equal(
+                '?url=' + encodeURIComponent('http://o.com/foo/?a'));
+          });
+    });
+
+    it('should not change SOURCE_URL if extra params are not available', () => {
```
Comment (lannka) [inline test/functional/test-url-replacements.js]: ... if servingType is not AD_LANDING_PAGE

**Comment by zhouyx** [comment]
@lannka I checked our existing outgoing link decoration. We don't append `=` to params w/o a value. I think we should be consistent and fix the behavior here as well.

**Comment by zhouyx** [comment]
@lannka  Let's sync quickly tomorrow on what the ideal behavior is and we can refactor the code a bit.

**Comment by zhouyx** [comment]
Created #17760 for improvement.

**Pull request merged**